### PR TITLE
added sample for code first grpc and documentation update

### PR DESCRIPTION
**Pull request by fiyazbinhasan**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/21965

***@Rick-Anderson Edit*** [Internal review URL](https://review.docs.microsoft.com/en-us/aspnet/core/grpc/code-first?view=aspnetcore-5.0&branch=pr-en-us-22128)

### CODE REVIEW

**Comment by fiyazbinhasan** [comment]
@Rick-Anderson, please review

**Comment by guardrex** [comment]
@fiyazbinhasan ... There looks to be two folders in use: `sample` and `snapshot_sample`, which is why the build is breaking.

While you're fixing that, could you use our pattern for maintaining samples by version? We use a `samples` folder, then a version folder (`5.x` in this case), and then the project folder (`GrpcGreeter` in this case). That way, when it comes time to have a 6.x sample, we create a `6.x` folder, copy the project folder into it, update the new project for 6.x, and add versioned links/sections to the topic. Later when 5.x goes out of support, we delete the whole `/5.x` folder and kill the topic's 5.x content. It's easy for us to maintained versioned samples.

Place the sample at the path ...

```
aspnetcore/grpc/code-first/samples/5.x/GrpcGreeter/
```

... and then update the links in the topic to point there.

One more tiny thing ... not that it's all that critical, but ... GH prefers that files end with a line feed on the last line of files. That's why you see those little red markers at the ends of the code files here on the diff. Can you provide a final CR/LF on those last lines? It doesn't really break anything if they aren't fixed, but it's our standard practice to make the 🙏 *GH server gods* 🙏 happy 😄.

**Comment by fiyazbinhasan** [comment]
Thanks @guardrex. I'll incorporate those changes soon. Better make the _GH server gods_ happy 😄

**Comment by fiyazbinhasan** [comment]
@guardrex I wasn't sure about the folders. So, I removed the `snapshot_sample` and used `region` directly in the source code for snippet usage. Let me know if I missed anything. thanks

**Comment by guardrex** [comment]
Thanks. AFAICT, this looks good. The build is happy now. Regions look fine. I'll put the internal doc link here for RA to inspect when he gets online ...

https://review.docs.microsoft.com/en-us/aspnet/core/grpc/code-first?view=aspnetcore-3.0&branch=pr-en-us-22128

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/grpc/code-first/samples/5.x/GrpcGreeterClient/GrpcGreeterClient.csproj [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (JamesNK) [inline aspnetcore/grpc/code-first/samples/5.x/GrpcGreeterClient/GrpcGreeterClient.csproj:5]:
```suggestion
    <TargetFramework>net5.0</TargetFramework>
```

File: aspnetcore/grpc/code-first/samples/5.x/Shared/Shared.Contracts.csproj [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (JamesNK) [inline aspnetcore/grpc/code-first/samples/5.x/Shared/Shared.Contracts.csproj:4]:
```suggestion
    <TargetFramework>net5.0</TargetFramework>
```

**Review by JamesNK — APPROVED** [review]

**Comment by guardrex** [comment]
Thanks @fiyazbinhasan! 🚀

**Pull request merged**